src/main/java/ahmad/processor/task/Add.java

```diff
@@ -0,0 +1,96 @@
+     * @throws DeadlineEmptyNameException   If name is empty.
+     * @throws EventEmptyNameException      If name is empty.
+     * @throws EventInvalidArgsException    If prompt/argument is invalid.
+     * @throws DeadlineInvalidTimeException If given time is invalid.
+     * @throws EventInvalidTimeException    If given time is invalid.
+     */
+    public static Response todo(String prompt) throws TodoEmptyNameException, DeadlineInvalidArgsException,
+            DeadlineEmptyNameException, EventEmptyNameException, EventInvalidArgsException,
+            DeadlineInvalidTimeException, EventInvalidTimeException {
+        final List<String> prompts = Arrays.asList(prompt.split("todo "));
+        if (prompts.size() < 2) {
+            throw new TodoEmptyNameException();
+        }
+        final Task newTask = Task.of(TaskType.Todo, prompts.get(1));
+        return Add.process(newTask);
+    }
+
+    /**
+     * Returns a response based on the prompt for a deadline command.
+     *
+     * @param prompt Prompt/argument for deadline command.
+     * @return Appropriate response for deadline command.
+     * @throws TodoEmptyNameException       If name is empty.
+     * @throws DeadlineInvalidArgsException If prompt/argument is invalid.
+     * @throws DeadlineEmptyNameException   If name is empty.
+     * @throws EventEmptyNameException      If name is empty.
+     * @throws EventInvalidArgsException    If prompt/argument is invalid.
+     * @throws DeadlineInvalidTimeException If given time is invalid.
+     * @throws EventInvalidTimeException    If given time is invalid.
+     */
+    public static Response deadline(String prompt) throws DeadlineEmptyNameException, DeadlineInvalidArgsException,
+            EventEmptyNameException, EventInvalidArgsException, TodoEmptyNameException,
+            DeadlineInvalidTimeException, EventInvalidTimeException {
+        final List<String> prompts = Arrays.asList(prompt.split("deadline "));
+        if (prompts.size() < 2) {
+            throw new DeadlineEmptyNameException();
+        }
+        final Task newTask = Task.of(TaskType.Deadline, prompts.get(1));
+        return Add.process(newTask);
+    }
+
+    /**
+     * Returns a response based on the prompt for an event command.
+     *
+     * @param prompt Prompt/argument for event command.
+     * @return Appropriate response for event command.
+     * @throws TodoEmptyNameException       If name is empty.
+     * @throws DeadlineInvalidArgsException If prompt/argument is invalid.
+     * @throws DeadlineEmptyNameException   If name is empty.
+     * @throws EventEmptyNameException      If name is empty.
+     * @throws EventInvalidArgsException    If prompt/argument is invalid.
+     * @throws DeadlineInvalidTimeException If given time is invalid.
+     * @throws EventInvalidTimeException    If given time is invalid.
+     */
+    public static Response event(String prompt) throws DeadlineInvalidArgsException, DeadlineEmptyNameException,

```

Comment:
Think about renaming the method so that it is a verb and clearly reflects the purpose of the method!